### ISSUE TITLE
Option to specify external configuration in a principled way

### ISSUE THREAD

**Issue by vjovanov** [graalvm/native-build-tools#197]
We need to provide a way to specify externally the reachability configuration for the dependencies of the project. The format of the external configuration folder is TBD 
 but it will follow a convention that allows mapping from a dependency's artifact ID to the config files. 

The first proposal for the DSL entry:

```groovy
graalvmNative {
    metadataRepository {
        enabled = true // default false until official repo release, true and optional later
        // path and version are mutually exclusive
        version = "0.0.1" // this would be the only mandatory field
        path = "/home/usr/dev/conf-repo|http://my.repo/|http://my.repo/repo-1.1.1.zip" // custom location or a local folder for development. default: empty which redirects to the configuration repo.
        ignoreDependencies = ["com.google.guava:guava"] // dependencies for which the repo is ignored. The version is omitted. 
    }
    binaries {
        main { 
        }
    }
}
```

In the first version, the configuration would be provided locally. Later we can use the GitHub releases, repository clones, or any other future mechanism.

**Comment by melix** [comment]
Fixed by #204 for Gradle. The name of the configuration block is currently `jvmReachabilityMetadataRepository`, which is a bit mouthful but accurate. We can shorten this to `metadataRepository` if needed.

```gradle
graalvmNative {
    metadataRepository {
        enabled = true // default false until official repo release, true and optional later
        // path and version are mutually exclusive
        version = "0.0.1" // this would be the only mandatory field
        path = "/home/usr/dev/conf-repo|http://my.repo/|http://my.repo/repo-1.1.1.zip" // custom location or a local folder for development. default: empty which redirects to the configuration repo.
        ignoreDependencies = ["com.google.guava:guava"] // dependencies for which the repo is ignored. The version is omitted. 
    }
    binaries {
        main { 
        }
    }
}
```

**Comment by sdeleuze** [comment]
+1 for using `metadataRepository`, in the context of being nested inside `graalvmNative` and properly documented that seems good enough to me.

**Comment by lazar-mitrovic** [comment]
Fixed by #204, #240 and #247.

**Comment by lazar-mitrovic** [comment]
Fixed by #204, #240 and #247.
